### PR TITLE
Allow escaping of texture names when passed as an argument to a modifier

### PR DESCRIPTION
**Pull request by sfan5**
As requested by @t4im :smiley:

### CODE REVIEW

File: src/client/tile.cpp [hunk 1]
```diff
@@ -1099,9 +1102,27 @@ video::IImage * Align2Npot2(video::IImage * image,
...
 #endif
 
+static std::string unescape_string(const std::string &str, const char esc = '\\')
+{
+	std::string out;
+	size_t pos = 0, cpos;
+	out.reserve(str.size());
+	while (1) {
+		cpos = str.find_first_of(esc, pos);
+		if(cpos == std::string::npos) {
```
Comment (est31) [inline src/client/tile.cpp]: space

**Comment by est31** [comment]
:+1:

**Comment by paramat** [comment]
Seems useful so i approve of the feature. I won't code review as you have the necessary approval and i'm not too good with this type of code (and i'm busy with other dev).

**Comment by t4im** [comment]
Just to show some examples, what this enables.

Automatically generated group inventory images (which is, why this came up):
group_dye
group_tree

Isometric inventory images:
isoart

These being texturepack aware and independent of texturepack resolution.

**Comment by paramat** [comment]
b77cee146b0029d377f1028b942857d062bc1374

**Pull request closed without merging**